tests/run-make/fmt-write-bloat/rmake.rs

```diff
@@ -12,24 +12,86 @@
+//! absence of additional `usize` formatting and padding related symbols.
+
+// ignore-tidy-linelength
 
 //@ ignore-cross-compile
 
-use run_make_support::artifact_names::bin_name;
+use std::path::Path;
+
 use run_make_support::env::std_debug_assertions_enabled;
-use run_make_support::rustc;
+use run_make_support::llvm::{llvm_filecheck, llvm_pdbutil};
 use run_make_support::symbols::object_contains_any_symbol_substring;
+use run_make_support::{bin_name, is_windows_msvc, rfs, rustc};
 
 fn main() {
-    rustc().input("main.rs").opt().run();
     // panic machinery identifiers, these should not appear in the final binary
     let mut panic_syms = vec!["panic_bounds_check", "Debug"];
     if std_debug_assertions_enabled() {
         // if debug assertions are allowed, we need to allow these,
         // otherwise, add them to the list of symbols to deny.
         panic_syms.extend_from_slice(&["panicking", "panic_fmt", "pad_integral", "Display"]);
     }
-    assert!(!object_contains_any_symbol_substring(bin_name("main"), &panic_syms));
+
+    let expect_no_panic_symbols = bin_name("expect_no_panic_symbols");
+    let expect_no_panic_symbols = Path::new(&expect_no_panic_symbols);
+    rustc().input("main.rs").output(&expect_no_panic_symbols).opt().run();
+
+    let expect_panic_symbols = bin_name("expect_panic_symbols");
+    let expect_panic_symbols = Path::new(&expect_panic_symbols);
+    rustc().input("main.rs").output(&expect_panic_symbols).run();
+
+    if is_windows_msvc() {
+        // FIXME(#143737): use actual DIA wrappers instead of parsing `llvm-pdbutil` textual output.
+
+        let expect_no_filecheck_pattern = r#"
+            CHECK: main
+            CHECK-NOT: {{.*}}Display{{.*}}
+            CHECK-NOT: {{.*}}panic_bounds_check{{.*}}
+        "#;
+        let expect_no_filecheck_path = Path::new("expect_no_panic_symbols_filecheck.txt");
+        rfs::write(&expect_no_filecheck_path, expect_no_filecheck_pattern);
+
+        let expect_no_panic_symbols_pdbutil_dump = llvm_pdbutil()
+            .arg("dump")
+            .arg("-publics")
+            .input(expect_no_panic_symbols.with_extension("pdb"))
+            .run();
+        llvm_filecheck()
+            .patterns(expect_no_filecheck_path)
+            .stdin_buf(expect_no_panic_symbols_pdbutil_dump.stdout_utf8())
+            .run();
+
+        // NOTE: on different platforms, they may not go through the same code path. E.g. on
+        // `i686-msvc`, we do not go through `panic_fmt` (only `panic`). So for the check here we
+        // only try to match for mangled `core::panicking` module path.
+        let expect_filecheck_pattern = r#"
+            CHECK: main
+            CHECK: {{.*}}core{{.*}}panicking{{.*}}
+            // _ZN4core3fmt3num3imp54_$LT$impl$u20$core..fmt..Display$u20$for$u20$usize...
+            CHECK: {{.*}}Display{{.*}}
+            CHECK-NOT: {{.*}}panic_bounds_check{{.*}}
+        "#;
```

Comment:
Remark: these are by _no means_ pretty, and in the future we almost certainly want our own DIA wrappers in `run_make_support` or somewhere. But I don't feel like implementing _that_ (at least in this PR), so I'll go with the dumb `llvm-pdbutil` approach for now.